src/kernel/sporadic.c

```diff
@@ -332,14 +343,6 @@ void refill_split_check(ticks_t usage)
 }
 
 
-static bool_t refill_unblock_check_mergable(sched_context_t *sc)
-{
-    ticks_t amount = refill_head(sc)->rAmount;
-    ticks_t tail = NODE_STATE_ON_CORE(ksCurTime, sc->scCore) + amount;
-    bool_t enough_time = refill_index(sc, refill_next(sc, sc->scRefillHead))->rTime <= tail;
-    return !refill_single(sc) && enough_time;
```

Comment:
What is requiring this change? refill_unblock_check_mergable was originally added to help the CParser which was unable to unfold the large conditional.  The value of enough_time was only used if !refill_single(sc) was true and there is no harm if the value is not needed, but it should create a shorter dependency chain and avoid the branch.

Reply:
That was causing assertion failures in code I had that checked calculated indexes for the head+count implementation. Whilst I removed those changes, I discussed whether the code here would be an issue for verification and they suggested it would probably throw a spanner in the works as-is when it got to refine as it would include an access to an entry in the list to which didn't have a corresponding entry in the aspec.

It's possible that the code as it was would be fine for refinement but the suggestion was that the array access was more likely to cause problems than not to.